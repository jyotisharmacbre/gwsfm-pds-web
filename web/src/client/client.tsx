import axios from 'axios';
import authentication from '@kdpw/msal-b2c-react';
import appConfig from '../helpers/config-helper';

const config = appConfig();
export const baseURL = config.REACT_APP_MIDDLETIER_URL;
export const baseAPI = axios.create({ baseURL });
export const userServiceURL = config.REACT_APP_GATEWAY_URL;
export const userServiceAPI = axios.create({ baseURL: userServiceURL });

const isTokenHandlerEnabled = (config = {}) => {
  return config.hasOwnProperty('tokenHandlerEnabled') &&
    !config['tokenHandlerEnabled']
    ? false
    : true;
};

const isErrorHandlerEnabled = (config = {}) => {
  return config.hasOwnProperty('errorHandlerEnabled') &&
    !config['errorHandlerEnabled']
    ? false
    : true;
};

const isSuccessHandlerEnabled = (config = {}) => {
  return config.hasOwnProperty('successHandlerEnabled') &&
    !config['successHandlerEnabled']
    ? false
    : true;
};

const requestHandler = request => {
  if (isTokenHandlerEnabled(request)) {
    // Add request token here
    const token = authentication.getAccessToken();
    request.headers['accessToken'] = token;
  }
  return request;
};

baseAPI.interceptors.request.use(request => requestHandler(request));
userServiceAPI.interceptors.request.use(request => requestHandler(request));

const errorHandler = error => {
  if (isErrorHandlerEnabled(error.config)) {
    // Handle errors
  }
  return Promise.reject({ ...error });
};

const successHandler = response => {
  if (isSuccessHandlerEnabled(response.config)) {
    // Handle responses
  }
  return response;
};

baseAPI.interceptors.response.use(
  response => successHandler(response),
  error => errorHandler(error)
);

userServiceAPI.interceptors.response.use(
  response => successHandler(response),
  error => errorHandler(error)
);
