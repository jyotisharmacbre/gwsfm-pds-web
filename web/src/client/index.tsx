export { baseAPI, userServiceAPI} from './client';
