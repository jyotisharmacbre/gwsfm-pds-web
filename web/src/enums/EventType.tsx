enum EventType {
  save,
  next,
  previous,
  none
}

export default EventType;
