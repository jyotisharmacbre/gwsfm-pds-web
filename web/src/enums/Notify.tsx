enum Notify {
  none,
  success,
  error
}

export default Notify;
