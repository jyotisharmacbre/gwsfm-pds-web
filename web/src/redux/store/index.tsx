import { applyMiddleware, createStore } from 'redux';
import { persistReducer, persistStore } from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import thunk from 'redux-thunk';

import allReducer from '../reducers';

const persistConfig = {
  key: 'root',
  storage
};

const persistedReducer = persistReducer(persistConfig, allReducer);

const middleware = applyMiddleware(thunk);
const store = createStore(persistedReducer, middleware);
const persistor = persistStore(store);
export { store, persistor };
