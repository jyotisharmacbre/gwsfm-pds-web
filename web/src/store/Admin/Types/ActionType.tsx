export enum ActionType {
	GET_PROJECT_PARAMETERS_SUCCESS = 'GET_PROJECT_PARAMETERS_SUCCESS',
	GET_PROJECT_PARAMETERS_ERROR = 'GET_PROJECT_PARAMETERS_ERROR'
}
