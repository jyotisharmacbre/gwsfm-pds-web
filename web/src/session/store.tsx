import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger'
import rootReducer from './reducers/RootReducer';

const middlewares: any[] = [];

middlewares.push(thunk);

if (process.env.NODE_ENV === `development`) {

  middlewares.push(logger);
}

export default function configureStore(initialState = {}) {
  return createStore(
    rootReducer,
    applyMiddleware(...middlewares)
  );
}