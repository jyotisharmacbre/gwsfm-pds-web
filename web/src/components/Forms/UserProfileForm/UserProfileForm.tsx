import React from 'react';
import {
  FontAwesomeIcon
} from '@fortawesome/react-fontawesome';
import {
  faUser
} from '@fortawesome/free-solid-svg-icons';

import { connect } from 'react-redux';
import { IState } from '../../../store/state';
import { Field, InjectedFormProps, formValueSelector, reduxForm } from 'redux-form';
import PdsFormSelect from '../../PdsFormHandlers/PdsFormSelect';
import { Validate } from '../../../helpers/fieldValidations';
import { FormattedMessage } from 'react-intl';
import { IUserPreferences } from '../../../store/UserPreferencesForm/Types/IUserPreferences';
import { ICurrency } from '../../../store/Lookups/Types/ICurrency';
import EventType from '../../../enums/EventType';
import { ILanguage } from '../../../store/Lookups/Types/ILanguage';
import { formatMessage } from '../../../Translations/connectedIntlProvider';

interface Props {
  onSubmitForm: (data: IUserPreferences, event: EventType) => void;
  redirectMenu: () => void;
  currencies: ICurrency;
  languages: ILanguage;
  displayName: string;
  displayEmail: string;

}

let UserProfileForm: React.FC<Props &
  InjectedFormProps<IUserPreferences, Props>> = (props: any, Name: any) => {

    const { handleSubmit, redirectMenu } = props;

    return (
      <form onSubmit={handleSubmit} noValidate={true}>
        <div>
          {/* START EDIT FORM SECTION */}

          <ul>
            <li>
              <a>
                <i>
                  <FontAwesomeIcon className="" icon={faUser} />
                </i>
                <p className="title_name">{props.displayName}</p>
                <span className="dsc">{props.displayEmail}</span>
              </a>
            </li>
            <li>
              <a>
                <i>
                  <FontAwesomeIcon className="" icon={faUser} />
                </i>
                <p className="title_name">{formatMessage('LABEL_PREFERED_LANGUAGE')}</p>
                <Field
                  name="languageId"
                  component={PdsFormSelect}
                  validate={Validate.required('LABEL_LANGUAGE')}
                  placeholderKey="PLACEHOLDER_LANGUAGE"
                  messageKey="MESSAGE_LANGUAGE"
                >
                  <FormattedMessage id="PLACEHOLDER_LANGUAGE">
                    {message => <option value="">{message}</option>}
                  </FormattedMessage>
                  {props.languages &&
                    props.languages.map((data: ILanguage, i: number) => {
                      return (
                        <option
                          key={data.languageID}
                          value={data.languageID}
                        >
                          {data.name}
                        </option>
                      );
                    })}
                </Field>
              </a>
            </li>
            <li>
              <a href="#">
                <i>
                  <FontAwesomeIcon className="" icon={faUser} />
                </i>
                <p className="title_name">{formatMessage('LABEL_PREFERED_CURRENCY')}</p>
                <Field
                  name="currencyId"
                  component={PdsFormSelect}
                  validate={Validate.required('LABEL_CURRENCY')}
                  placeholderKey="PLACEHOLDER_CURRENCY"
                  messageKey="MESSAGE_CURRENCY"
                >
                  <FormattedMessage id="PLACEHOLDER_CURRENCY">
                    {message => <option value="">{message}</option>}
                  </FormattedMessage>
                  {props.currencies &&
                    props.currencies.map((data: ICurrency, i: number) => {
                      return (
                        <option
                          key={data.currencyId}
                          value={data.currencyId}
                        >
                          {data.currencyName}
                        </option>
                      );
                    })}
                </Field>
              </a>
            </li>
          </ul>


          <div className='link_group'>
            <a href="#" data-test="save" onClick={handleSubmit(values => props.onSubmitForm(values))}><FormattedMessage id="BUTTON_SAVE" /></a>
            <span>|</span>
            <a href="#" data-test="cancel" onClick={redirectMenu}>{formatMessage('BUTTON_CANCEL')}</a>
          </div>

          {/* END EDIT FORM SECTION */}
        </div>
      </form>
    );
  }


const mapStateToProps = (state: IState) => {
  return {
    initialValues: state.userPreferences.preferences,
    currencyId: selector(state, 'currencyId'),
    languageId: selector(state, 'languageId'),
  }
}


const form = reduxForm<IUserPreferences, Props>({
  form: 'UserProfileForm',
  enableReinitialize: true
})(UserProfileForm);

const selector = formValueSelector('UserProfileForm');


export default connect(mapStateToProps)(form)