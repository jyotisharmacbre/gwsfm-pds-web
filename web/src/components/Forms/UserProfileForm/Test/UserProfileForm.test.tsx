import React from 'react';
import { mount, ShallowWrapper } from 'enzyme';
import { Provider } from 'react-redux';
import { store } from '../../../../store';
import UserProfileForm from '../UserProfileForm';
import { IntlProvider } from 'react-intl';
import translations from '../../../../Translations/translation';
import * as connectedIntlProvider from './../../../../Translations/connectedIntlProvider';
import Notify from '../../../../enums/Notify';
import EventType from '../../../../enums/EventType';
import UserPreferencesReducer from '../../../../store/UserPreferencesForm/Reducer';
import { ActionType } from '../../../../store/UserPreferencesForm/Types/ActionType';
import nock from 'nock';
import { baseURL } from '../../../../client/client';
import { IUserPreferencesState } from '../../../../store/UserPreferencesForm/Types/IUserPreferencesState';

const initialState: IUserPreferencesState = {
  preferences: {
    userPreferenceId: '',
    languageId: 1,
    languageName: 'en',
    currencyId: 1,
    currencySymbol: '$',
    currencyName: 'Dollar'
  },
  error: null,
  loading: false,
  notify: Notify.none,
  event: EventType.none
};

nock(baseURL)
  .post('/api/Users/addUserPreferences')
  .reply(200, 'Preferences added successfully');

nock(baseURL)
  .put('/api/Users/updateUserPreferences')
  .reply(201, 'Preferences updated successfully');

nock(baseURL)
  .get('/api/Users/getUserPreferences')
  .reply(200, { languageName: 'en', languageId: 1 });

describe('UserProfileForm Fields', () => {
  let handleEvent = () =>{};
  let closePanel = () =>{};
  let wrapper: any;
  const props: any = {
    handleSubmit: jest.fn()
  };
  beforeEach(() => {
    const formatMessage = jest.mock(
      './../../../../Translations/connectedIntlProvider'
    );
    jest
      .spyOn(connectedIntlProvider, 'formatMessage')
      .mockImplementationOnce(() => {
        return 'intlmessage';
      });

    wrapper = mount(
      <Provider store={store}>
        <IntlProvider locale="en" messages={translations['en'].messages}>
        <UserProfileForm onSubmitForm={handleEvent}
                          redirectMenu = {closePanel}
                          currencies={props.currencies}
                          languages={props.languages}
                          displayName= 'userName'
                          {...props}
                        />
        </IntlProvider>
      </Provider>
    );
  });
  it('Defines the component', () => {
    expect(wrapper).toBeDefined();
  });

  it('Defines the Form', () => {
  let form = wrapper.find('[form="UserProfileForm"]').first();

      expect(form).toHaveLength(1);
  });

   describe('Defines form fields', () => {
    let field: ShallowWrapper;

    describe('languageId field', () => {
      beforeEach(() => {
        field = wrapper.find('select[name="languageId"]').first();
        wrapper.setProps({
          currencies: [
            { languageId: 1, languageName: 'english' }
          ],
          languageId: 1
        });
      });
      it('Should renders languageId field', () => {
        expect(field.render());
      });
      it('Shows error when languageId is set to blank', () => {
        field.simulate('blur');
        const errorBlock = wrapper.find('.text-danger');
        expect(errorBlock).toHaveLength(1);
      });
    });

    describe('currencyId field', () => {
      beforeEach(() => {
        field = wrapper.find('select[name="currencyId"]').first();
        wrapper.setProps({
          currencies: [
            { currencyId: 1, currencySymbol: '$', currencyName: 'dollar' }
          ],
          currencyId: 1
        });
      });
      it('Should renders currencyId field', () => {
        expect(field.render());
      });
      it('Shows error when currencyId is set to blank', () => {
        field.simulate('blur');
        const errorBlock = wrapper.find('.text-danger');
        expect(errorBlock).toHaveLength(1);
      });
    });



    describe('Save button', () => {
      beforeEach(() => {
        field = wrapper.find('a[data-name="save"]').first();
      });
      it('Should renders save button', () => {
        expect(field).toHaveLength(1);
      });
    });

    describe('UserProfile form reducer', () => {
      it('should handle get getUserPreferences successfully', () => {
        const getUserPreferencesAction: any = {
          type: ActionType.USER_PREFERENCES_GET_SUCCESS,
          payload: { languageName: 'french', languageId: 2 }
        };
        expect(
          UserPreferencesReducer(initialState, getUserPreferencesAction)
        ).toMatchSnapshot();
      });

      it('should handle edit userPreferences successfully', () => {
        const editUserPreferencesAction: any = {
          type: ActionType.USER_PREFERENCES_FORM_EDIT_SUCCESS,
          payload: { languageName: 'french', languageId: 2 }
        };
        expect(
          UserPreferencesReducer(initialState, editUserPreferencesAction)
        ).toMatchSnapshot();
      });

      it('should handle add userPreferences successfully', () => {
        const addUserPreferencesAction: any = {
          type: ActionType.USER_PREFERENCES_FORM_ADD_SUCCESS
        };
        expect(
          UserPreferencesReducer(initialState, addUserPreferencesAction)
        ).toMatchSnapshot();
      });

      it('should handle userPreferenceUpdateError with and return initialState', () => {
        const userPreferenceUpdateError: any = {
          type: ActionType.USER_PREFERENCES_FORM_ERROR,
          error: { success: false }
        };
        expect(
          UserPreferencesReducer(initialState, userPreferenceUpdateError)
        ).toMatchSnapshot();
      });
    });
  });
});
