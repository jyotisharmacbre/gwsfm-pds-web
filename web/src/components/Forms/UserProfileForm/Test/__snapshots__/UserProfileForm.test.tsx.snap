// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfileForm Fields Defines form fields UserProfile form reducer should handle add userPreferences successfully 1`] = `
Object {
  "error": null,
  "event": undefined,
  "loading": false,
  "notify": 1,
  "preferences": Object {
    "currencyId": 1,
    "currencyName": "Dollar",
    "currencySymbol": "$",
    "languageId": 1,
    "languageName": "en",
    "userPreferenceId": "",
  },
}
`;

exports[`UserProfileForm Fields Defines form fields UserProfile form reducer should handle edit userPreferences successfully 1`] = `
Object {
  "error": null,
  "event": undefined,
  "loading": false,
  "notify": 1,
  "preferences": Object {
    "currencyId": 1,
    "currencyName": "Dollar",
    "currencySymbol": "$",
    "languageId": 1,
    "languageName": "en",
    "userPreferenceId": "",
  },
}
`;

exports[`UserProfileForm Fields Defines form fields UserProfile form reducer should handle get getUserPreferences successfully 1`] = `
Object {
  "error": null,
  "event": undefined,
  "loading": false,
  "notify": 0,
  "preferences": Object {
    "languageId": 2,
    "languageName": "french",
  },
}
`;

exports[`UserProfileForm Fields Defines form fields UserProfile form reducer should handle get resetUserPreferencesState successfully 1`] = `
Object {
  "error": null,
  "event": 3,
  "loading": false,
  "notify": 0,
  "preferences": Object {
    "currencyId": 1,
    "currencyName": "Dollar",
    "currencySymbol": "$",
    "languageId": 1,
    "languageName": "en",
    "userPreferenceId": "",
  },
}
`;

exports[`UserProfileForm Fields Defines form fields UserProfile form reducer should handle userPreferenceUpdateError with and return initialState 1`] = `
Object {
  "error": Object {
    "success": false,
  },
  "event": 3,
  "loading": false,
  "notify": 2,
  "preferences": Object {
    "currencyId": 1,
    "currencyName": "Dollar",
    "currencySymbol": "$",
    "languageId": 1,
    "languageName": "en",
    "userPreferenceId": "",
  },
}
`;

exports[`UserProfileForm Fields Defines form fields UserProfile form reducer should return userPreferencesFormAdd action 1`] = `
Object {
  "event": 0,
  "payload": Object {
    "currencyId": 1,
    "currencyName": "Dollar",
    "currencySymbol": "$",
    "languageId": 1,
    "languageName": "en",
    "userPreferenceId": "",
  },
  "type": "USER_PREFERENCES_FORM_ADD_SUCCESS",
}
`;

exports[`UserProfileForm Fields Defines form fields UserProfile form reducer should return userPreferencesFormEdit action 1`] = `
Object {
  "event": 0,
  "payload": Object {
    "currencyId": 1,
    "currencyName": "Dollar",
    "currencySymbol": "$",
    "languageId": 1,
    "languageName": "en",
    "userPreferenceId": "",
  },
  "type": "USER_PREFERENCES_FORM_EDIT_SUCCESS",
}
`;

exports[`UserProfileForm Fields Defines form fields UserProfile form reducer should return userPreferencesGet action 1`] = `
Object {
  "payload": Object {
    "languageId": 1,
    "languageName": "en",
  },
  "type": "USER_PREFERENCES_GET_SUCCESS",
}
`;

exports[`UserProfileForm Fields Defines form fields should return resetUserPreferencesState action 1`] = `undefined`;

exports[`UserProfileForm Fields Defines form fields should return resetUserPreferencesState action 2`] = `
Object {
  "payload": "error",
  "type": "USER_PREFERENCES_FORM_ERROR",
}
`;
